raiden/network/proxies/token_network.py

```diff
@@ -1085,3 +1055,133 @@ def _check_for_outdated_channel(
                 f'new={onchain_channel_identifier}',
             )
+
+    def _get_channel_state(self, participant1, participant2, channel_identifier):
+        try:
+            channel_data = self.detail_channel(participant1, participant2, channel_identifier)
+        except RaidenUnrecoverableError:
+            return False
+
+        if not isinstance(channel_data.state, typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        return channel_data.state
+
+    def _check_channel_state_for_close(self, participant1, participant2, channel_identifier):
+        channel_state = self._get_channel_state(
+            participant1=participant1,
+            participant2=participant2,
+            channel_identifier=channel_identifier,
+        )
+
+        if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            raise RaidenUnrecoverableError(
+                f'Channel between participant {participant1} '
+                f'and {participant2} does not exist',
+            )
+        # Deposit was prohibited because the channel is settled
+        elif channel_state == ChannelState.SETTLED:
+            raise RaidenUnrecoverableError(
+                'Channel is not in an settled state',
+            )
+        # Deposit was prohibited because the channel is closed
+        elif channel_state == ChannelState.CLOSED:
+            raise RaidenRecoverableError(
+                'Channel is not in an open state',
+            )
+
+    def _check_channel_state_for_deposit(
+            self,
+            participant1,
+            participant2,
+            channel_identifier,
+            deposit_amount,
+    ):
+        participant_details = self.detail_participants(
+            participant1,
+            participant2,
+            channel_identifier,
+        )
+
+        if participant_details.our_details.deposit > deposit_amount:
+            raise RaidenUnrecoverableError('Deposit amount decreased')
+
+        channel_state = self._get_channel_state(
+            participant1=self.node_address,
+            participant2=participant2,
+            channel_identifier=channel_identifier,
+        )
+        # Check if deposit is being mode on a nonexistent channel
+        if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            raise RaidenUnrecoverableError(
+                f'Channel between participant {participant1} '
+                f'and {participant2} does not exist',
+            )
+        # Deposit was prohibited because the channel is settled
+        elif channel_state == ChannelState.SETTLED:
+            raise RaidenUnrecoverableError(
+                'Channel is settled',
+            )
+        # Deposit was prohibited because the channel is closed
+        elif channel_state == ChannelState.CLOSED:
+            raise RaidenRecoverableError(
+                'Channel is not in an open state',
```

Comment:
Here too, the state of the channel could be closed/settled/removed, what about something like `Deposit failed because channel closed meanwhile`?